sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/CacheTableExec.scala

```diff
@@ -94,14 +94,21 @@ case class CacheTableAsSelectExec(
   override lazy val relationName: String = tempViewName
 
   override lazy val planToCache: LogicalPlan = {
+    // CacheTableAsSelectExec.query is not resolved yet (e.g., not a child of CacheTableAsSelect)
+    // in order to skip optimizing it; note that we need to pass an analyzed plan to
+    // CreateViewCommand for the cache to work correctly. Thus, the query is analyzed below.
```

Comment:
Can we further clean this up by creating `CreateViewStatement` instead of `CreateViewCommand` below?